packages/lib/BaseApplication.ts

```diff
@@ -500,6 +501,13 @@ export default class BaseApplication {
 				await sideEffects[key]();
 			}
 		}
+
+		setupProxySettings({
+			maxConcurrentConnections: Setting.value('sync.maxConcurrentConnections'),
+			proxyTimeout: Setting.value('net.proxyTimeout'),
+			proxyEnabled: Setting.value('net.proxyEnabled'),
+			proxyUrl: Setting.value('net.proxyUrl'),
+		});
```

Comment:
Just one thing here, is that we want to run this only when a relevant setting is changed.

For that you can use the same pattern as above:

In the `sideEffects` object, you add a key `net.proxyEnabled` for example, which does the `setupProxySettings()` call. Then below, you add the additional keys: `sideEffects['sync.maxConcurrentConnections'] = sideEffects['net.proxyEnabled']`, and that for each of the three other required settings.